### ISSUE TITLE
Provide hsc2hs replacement

### ISSUE THREAD

**Issue by chpatrick** [fpco/inline-c#31]
It would be really cool if inline-c could do some stuff that we use hsc2hs for currently, ie. sizeOf and friends. It would remove the need to include stuff twice and would be much nicer with template haskell than the print-out-the-whole-source-with C approach.

**Comment by bitonic** [comment]
Well, you can already do

```
[C.exp| sizeof (int) |]
```

The problem is that it will be executed each time rather than known statically.

Do you mean that we should be able to execute C code statically and embed results (a-la TemplateHaskell, but with C)?

**Comment by chpatrick** [comment]
Yes, that's what I meant. It would be nice because it would ensure that it runs in the same C environment as the inline-c code, and it would be much more extensible than hsc's macro hacks.

**Comment by ekmett** [comment]
It isn't clear to me that this is very possible.

`hsc2hs` can do things like splice into import lists and even emit additional haskell language directives above the very module signature, because all it is doing is building a c program that will print a haskell program. It does its work on source text, not `template-haskell`.
